promising/promising_context.py

```
@@ -736,22 +842,47 @@ def _resolve_thread_pool(
+    *is* entered, observers awaiting the future may be woken up before
+    the surrounding ``with`` block exits. Closing the context first
+    ensures that anyone who sees the future as done also sees its
+    context as closed — in particular, a parent's
+    ``await_children()`` loop will not pick this child up again on its
+    next iteration.
+    """
 
-    def _call_soon_threadsafe(self, callback: Callable[[], Any]) -> None:
-        if not self._ctx_loop.is_running():
-            raise NoRunningEventLoopError(f"The event loop of {self} is not running")
+    def __init__(self, **kwargs: Any) -> None:
```

Ruff (0.15.9)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `**kwargs`

(ANN401)